### ISSUE TITLE
[BUG] Cannot start `create-daml-app` using Node v17.4.0

### ISSUE THREAD

**Issue by darko-da** [digital-asset/daml#13062]
# Affected Daml version

1.18.1

# Bug description

When starting the UI of the default `create-daml-app` application using `node v17.4.0` the following error pops up:
```
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:67:19)
    at Object.createHash (node:crypto:135:10)
    at NormalModule._initBuildHash (E:\test\myapp\ui\node_modules\webpack\lib\NormalModule.js:417:16)
    at E:\test\myapp\ui\node_modules\webpack\lib\NormalModule.js:503:5
    at E:\test\myapp\ui\node_modules\webpack\lib\NormalModule.js:358:12
    at E:\test\myapp\ui\node_modules\loader-runner\lib\LoaderRunner.js:373:3
    at iterateNormalLoaders (E:\test\myapp\ui\node_modules\loader-runner\lib\LoaderRunner.js:214:10)
    at iterateNormalLoaders (E:\test\myapp\ui\node_modules\loader-runner\lib\LoaderRunner.js:221:10)
```

By changing the start script in `package.json` to 
```
"scripts": {
    "start": "react-scripts --openssl-legacy-provider start",
    ...
}
```
The error goes away.

# To reproduce

Prerequisite: `Node v17.4.0`
1. Create new daml project using the `create-daml-app` template
2. Run `npm install` in the UI subfolder
3. Run `npm start`
4. see error

**Comment by ray-roestenburg-da** [comment]
If this can be upgraded to work with Node v17.4.0 or higher, we should target Node 18 (which is scheduled for LTS https://nodejs.org/en/about/releases/).

**Comment by chunlokling-da** [comment]
@ray-roestenburg-da So Currently there is no issue with node v16. However, if we want to make it work with v18, we will need to do some `` but it will likely break node v16. Dose it make sense to decide whether we only support a version?

There is a workaround: adding NODE_OPTIONS=--openssl-legacy-provider to "scripts" but it seems hacky.

**Comment by ray-roestenburg-da** [comment]
@chunlokling-da The advice in [node releases](https://nodejs.org/en/about/releases/) is to support the active LTS (which is the v16 right now, EOL 2023-09-11) and maintenance LTS (which is v14 right now,  EOL 2023-04-30) version for production usage. [EOL releases](https://github.com/nodejs/Release/#end-of-life-releases) includes v17 (EOL 2022-06-01), so we should not support this at all.

We should ask users to use v16 instead. v18 starts as active LTS version at 2022-10-25. We should postpone changing the create-daml-app to work with v18 (forcing the audit) until it becomes the active LTS version, and at that time tell users to move from v16 to v18. Right now (for this ticket) we should simply document that the problem is fixed with the `NODE_OPTIONS=--openssl-legacy-provider` workaround for v17, and that this version has gone EOL, pointing to the nodejs [EOL releases](https://github.com/nodejs/Release/#end-of-life-releases) page.
